### PR TITLE
DOC-3619: Added second contracts tutorial.

### PR DESCRIPTION
**Pull request by taylorharvey-r3**

### CODE REVIEW

File: content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract-2.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+class BoardingTicketContract : Contract {
+
+}
+```
+
+### Connect the `BoardingTicketContract` to the `BoardingTicket` state
+
+After creating the contract class in a CorDapp, you must connect the contract to its correlating state. Add the `@BelongsToContract` annotation *in the state class* to establish the relationship between a state and a contract. Without this, your state does not hold a relationship to the contract that is used to verify it.
+
+1. Open your `BoardingTicket` class.
```
Comment (peterli-r3) [inline content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract-2.md:65]: This line is redundant

File: content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract-2.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+* For the `CreateTicket` command:
+
+  * The transaction should only output one `BoardingTicket` state.
+  * The output `BoardingTicket` state should have a clear description of the space trip.
+  * The output `BoardingTicket` state should have a launch date later then the creation time.
+
+* For the `RedeemTicket` command:
+
+  * The transaction should consume two states.
```
Comment (fenryka) [inline content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract-2.md:29]: might want to say why

File: content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract.md [hunk 3]
```diff
@@ -206,94 +216,35 @@ class MarsVoucherContract : Contract {
...
-                null
-            }
-        }
-    }
 
     // Used to indicate the transaction's intent.
     interface Commands : CommandData {
-        class CreateTicket : Commands
-        class RedeemTicket : Commands
+        //In our hello-world app, We will have two commands.
```
Comment (fenryka) [inline content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract.md:235]: it's not a hello world app is it? its the mars trip app

File: content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-state-2.md [hunk 4]
```diff
@@ -42,7 +42,7 @@ Include these variables in the `BoardingTicket` data class:
 * `description`: Trip information. Use type `String`.
 * `marsExpress`: The space travel company issuing the ticket. Use type `Party`.
 * `owner`: The party exchanging the voucher for the ticket. Use type `Party`.
-* `daysUntilLaunch`: The number of days until the launch date. Use type `int`.
+* `launchDate`: The number of days until the launch date. Use type `LocalDate`.
```
Comment (fenryka) [inline content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-state-2.md:45]: launch date isn't the number of days till the launch, its a date :)

File: content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-state-2.md [hunk 5]
```diff
@@ -95,7 +95,7 @@ data class BoardingTicket(
 There are several ways to return your parameters in a JSON string. This tutorial shows you one method, but you are not restricted to using this specific method in Corda.
 {{< /note >}}
 
-1. Create a data transfer object that encapsulates the data of your `BoardingTicket` state—`BoardingTicketDto`. Include the same variables as the `BoardingTicket` class (`description`, `marsExpress`, `owner`, and `daysUntilLaunch`) and mark all variable types as `String`.
+1. Create a data transfer object that encapsulates the data of your `BoardingTicket` state—`BoardingTicketDto`. Include the same variables as the `BoardingTicket` class (`description`, `marsExpress`, `owner`, and `launchDate`) and mark all variable types as `String`.
```
Comment (fenryka) [inline content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-state-2.md:98]: we don't mark variable types as string. you define a variable as a specific type, in this case string. If we're talking about the participant's creation then we're converting variables to strings to pass to that

File: content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract.md [hunk 6]
```diff
@@ -206,94 +216,35 @@ class MarsVoucherContract : Contract {
             is BoardingTicketContract.Commands.RedeemTicket-> requireThat {
                 //Transaction verification will happen in BoardingTicket Contract
             }
+            is Commands.Transfer -> requireThat {
```
Comment (fenryka) [inline content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract.md:219]: shouldn't we also check that its the current holder initiating the transfer

**Review by fenryka — APPROVED** [review]
It's a lot better, added some minor comments

File: content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract.md (hunk 6 above)
Comment (taylorharvey-r3) [reply content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract.md:219]: @peterli-r3 What do you think?

File: content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-state-2.md (hunk 4 above)
Comment (taylorharvey-r3) [reply content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-state-2.md:45]: Good catch!

File: content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract.md (hunk 6 above)
Comment (peterli-r3) [reply content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract.md:219]: Sure. Will add it to the code.

Comment (peterli-r3) [reply content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract.md:219]: Initiator check added in the GiftVoucherToFriend Flow.

Comment (taylorharvey-r3) [reply content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract.md:219]: Okay I'll address this in a separate PR. Thanks.

**Review by erinking-r3 — COMMENTED** [review]
a few minor comments, but great work!

File: content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract-2.md [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+- tutorial
+- cordapp
+title: Write a complex contract
+---
+
+Now that you've written your first contract, write the `BoardingTicketContract`.
+
+The `BoardingTicketState` will be used on two occasions: when the ticket is created by Mars Express and when the ticket is redeemed by the customer. This means that the `BoardingTicketContract` should have a `Commands` interface that carries two commands corresponding to the contract's two intentions.
+
+The rules inside the `requireThat` Corda DSL helper method are:
```
[the remark this answers is not in the document]
Comment (taylorharvey-r3) [reply content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract-2.md:19]: You're not the worst at all! This is good.

File: content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract-2.md [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+
+3. In the **New Kotlin Class/File** window, select **Class** and name the file `BoardingTicketContract`.
+
+4. Open the file.
+
+### Create the contract class
+
+As noted when creating the `MarsVoucherContract`, Corda states typically have a corresponding contract class to document the rules/policy of that state when used in a transaction. To declare the contract class:
+
+Add the class name `BoardingTicketContract` that implements the `Contract` class.
```
[the remark this answers is not in the document]
Comment (taylorharvey-r3) [reply content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract-2.md:49]: I definitely know this.. Oops. Thanks.

**Review by erinking-r3 — APPROVED** [review]

**Pull request merged**